### ISSUE TITLE
Whether there have pre-train code and re-annotating scripts for CheckM2 model on custom database?

### ISSUE THREAD

**Issue by zhaorui-bi** [chklovski/CheckM2#153]
So greate work for checkm2. 

This is issue based on #114, I intend to use CheckM2 based on my custom database but in vain. And I saw the issue #114, it shoud re-annotate and re-train the CheckM2 models. However, it seems that it has not provided in the offical GitHub rep.

Whether there have pre-train code for training CheckM2 models for custom database. And the re-annotating scripts whether checkm2 team can kindly provide them. I can not find them in the official GitHub rep.

**Comment by chklovski** [comment]
You can adapt the code and scripts from the checkm2 updater repo: https://github.com/chklovski/checkm2_supplementary

Keep in mind that 1) this will take a substantial amount of effort if you are regenerating the entire training set (you need to re-generate hundreds of thousands of simulated genomes, and annotate them with KEGG Ids using DIAMOND, which is a substantial amount of compute), and 2) this will have no relation to the official CheckM2 release or publication.

Can I ask what the purpose is? Generally there is no point using anything other than complete genomes for synthetic genome generation (because you need to know exactly their completeness/contamination, which you can only do if you're starting with a 100% complete, 0% contaminated genome), and CheckM2 already includes most existing complete RefSeq genomes (currently up to GTDB release 207, upcoming version up to release 226)?

**Comment by zhaorui-bi** [comment]
Thanks great for your so quick response. I will try to follow it!

Main purpose for me is to try to use pytorch framework rather than tensorflow.keras owing to some python version conflict with our project BASALT https://github.com/EMBL-PKU/BASALT new version. Another point is I still confused with chekcm and checkm2 which is better on world real data.

**Comment by chklovski** [comment]
CheckM2 shows better accuracy than CheckM1 on real-world data, particularly lineages that are not part of the training set (see publication for comparison) but I cannot guarantee anything if you re-train the models.

**Comment by zhaorui-bi** [comment]
Thanks greatly for your kind response. And I will check them in checkm2 paper in details! Thanks you again!
